articles/tutorials/advanced/2d_shaders/02_hot_reload/index.md

```diff
@@ -0,0 +1,280 @@
+
+[!code-sh[](./snippets/snippet-2-05.sh)]
+
+Now, when you change a _`.fx`_ file, all of the content files are rebuilt into `.xnb` files.
+
+However, the `.xnb` files are not being copied from the `Content/bin` folder to _DungeonSlime_'s runtime folder. The `.xnb` files are only copied during the full MSBuild of the game. The `IncludeContent` target on its own does not have all the context it needs to know how to copy the files in the final game project. To solve this, we need to introduce a new `<Target>` that copies the final `.xnb` files into _DungeonSlime_'s runtime folder. 
+
+The existing `MonoGame.Content.Builder.Task` system knows what the files are, so we can re-use properties defined in the MonoGame package.
+Add this `<Target>` block to your `.csproj` file:
+
+[!code-xml[](./snippets/snippet-2-06.xml)]
+
+Now, instead of calling the `IncludeContent` target directly, change your terminal command to invoke the new `BuildAndCopyContent` target:
+
+[!code-sh[](./snippets/snippet-2-07.sh)]
+
+If you delete the `DungeonSlime/bin/Debug/net8.0/Content` folder, make an edit to a `.cs` file and save, you should see the `DungeonSlime/bin/Debug/net8.0/Content` folder be restored.
+
+The next step is to only invoke the target when `.fx` files are edited instead of `.cs` files. These settings can be configured with custom MSBuild item configurations. Open the `DungeonSlime.csproj` file and add this `<ItemGroup>` to specify configuration settings:
+
+[!code-xml[](./snippets/snippet-2-08.xml)]
+
+Now when you re-run the command from earlier, it will only run the `IncludeContent` target when `.fx` files have been changed. All edits to `.cs` files are ignored. Try adding a blank line to the `grayscaleEffect.fx` file, and notice the `dotnet watch` process re-build the content.
+
+However, if you would like to use `dotnet watch` for anything else in your workflow, then the configuration settings are too aggressive, because they will be applied _all_ invocations of `dotnet watch`. The `ItemGroup` can be optionally included when a certain condition is met. We will introduce a new MSBuild property called `OnlyWatchContentFiles`:
+
+[!code-xml[](./snippets/snippet-2-09.xml?highlight=1)]
+
+And now when `dotnet watch` is invoked, it needs to specify the new parameter:
+
+[!code-sh[](./snippets/snippet-2-10.sh)]
+
+The command is getting long and hard to type, and if we want to add more configuration, it will likely get even longer. Instead of invoking `dotnet watch` directly, it can be run as a new `<Target>` MSBuild step. Add this `<Target>` to your `DungeonSlime.csproj` file:
+
+[!code-xml[](./snippets/snippet-2-11.xml)]
+
+And now from the terminal, run the following `dotnet build` command:
+
+[!code-sh[](./snippets/snippet-2-12.sh)]
+
+
+We now have a way to dynamically recompile shaders on file changes and copy the `.xnb` files into the game folder! There are a few final adjustments to make to the configuration. 
```

Comment:
Worth adding an admonition here to note that changes to the CS files are no longer watched and do not recompile (at least for me), I changed the example above for the title screen and nothing happened.

If this is expected, then highlight it to the reader.

Also annoying it no longer launches the game when the command is run, but that is dotnet for you I guess.

Worth highlighting how the reader is now supposed to launch the game in debug, debug -> start new instance, or simply run the compiled exe?

Reply:
I tested running the same command from this tutorials chapter source and it worked?   But following the instructions from the final chapter of the 2D tutorial did not?